File: client/python/mesh/kinds/registration.py
#
# Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
# TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
#
#
from typing import Any, List, Dict

from mesh.kinds.reference import Reference
from mesh.kinds.service import Service
from mesh.macro import index, serializable

Binding = "Binding"
MPS = "MPS"
Forward = "Forward"
Proxy = "Proxy"


@serializable
class Registration:
    METADATA = "metadata"
    PROXY = "proxy"
    SERVER = "server"
    COMPLEX = "complex"

    @index(0)
    def instance_id(self) -> str:
        return ''

    @index(5)
    def name(self) -> str:
        return ''

    @index(10)
    def kind(self) -> str:
        return ''

    @index(15)
    def address(self) -> str:
        return ''

    @index(20)
    def content(self) -> Any:
        return None

    @index(25)
    def timestamp(self) -> int:
        return 0

    @index(30)
    def attachments(self) -> Dict[str, str]:
        return {}


class Resource:

    @index(0)
    def references(self) -> List[Reference]:
        return []

    @index(5)
    def services(self) -> List[Service]:
        return []

File: client/javascript/src/kinds/registration.ts
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

import {index} from "@/macro/idx";

export class Registration {

    @index(0)
    public instance_id: string = "";
    @index(5)
    public name: string = "";
    @index(10)
    public kind: string = "";
    @index(15)
    public address: string = "";
    @index(20)
    public content: any = "";
    @index(25)
    public timestamp: number = 0;
    @index(30)
    public attachments: Map<string, string> = new Map<string, string>();

}